### DISCUSSION TITLE
Using popper in typescript -> rollup -> js

### DISCUSSION THREAD

**Discussion by Eonasdan** [floating-ui/floating-ui#1257] [Q&A] [answered]
Hello, thanks for making this great library.

I'm currently working on a new version of my date time picker. I'm trying to remove bootstrap as a hard dependency and previously, I relied on their dropdown menu (which uses popper under the covers).

I'm rewriting my picker in typescript and I'm having a bit of trouble getting popper setup.

[I have](https://github.com/Eonasdan/tempus-dominus/blob/v6-alpha1/src/js/tempus-dominus.ts):

```
import { createPopper } from '@popperjs/core/lib/createPopper';

this.popperInstance = createPopper(document.querySelector('#popcorn'), element, {
            modifiers: [
                {
                    name: 'offset',
                    options: {
                        offset: [0, 8],
                    },
                },
            ],
        });
```

I then  use rollup to transpile all the ts files to js but when I run the output, I get `Cannot read property 'createPopper' of undefined`. I have a script tag for popper via cdnjs. Am I just missing something obvious here?

**Comment by FezVrasta** [comment]
Any reason why are you importing from `@popperjs/core/lib/createPopper` rather than from `@popperjs/core`?

**Reply by Eonasdan** [reply]
no reason other then trying to get it work. Either way, I had the same error. I thought maybe ts was only pretending it didn't have a problem with the latter lol

**Reply by Eonasdan** [reply]
I just noticed that rollup was outputting this:

```
(!) Unresolved dependencies
https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency
@popperjs/core (imported by src\js\tempus-dominus.ts)
(!) Missing global variable name
Use output.globals to specify browser global variable names corresponding to external modules
@popperjs/core (guessing 'core')
```

so maybe this is a rollup config issue and not a popper issue.

**Reply by FezVrasta** [reply]
Yes it's a config issue.

**Reply by Eonasdan** [reply]
I got it sorted.

```
import typescript from '@rollup/plugin-typescript';

export default {
    output: {        
        globals: {
            '@popperjs/core': 'Popper'
        }
    },
    external: [ '@popperjs/core' ],
    plugins: [
        typescript()
    ],
};
```

**Comment by Eonasdan** [comment, marked as answer]
I got it sorted.

```
import typescript from '@rollup/plugin-typescript';

export default {
    output: {        
        globals: {
            '@popperjs/core': 'Popper'
        }
    },
    external: [ '@popperjs/core' ],
    plugins: [
        typescript()
    ],
};
```
